### PR TITLE
Bazel CAS batch operations

### PR DESCRIPTION
**Pull request by dgdonut**
Enables BatchUpdateBlobs, BatchReadBlobs. FindMissingBlobs now parallelized.

Adds support for above APIs with parallelized access to underlying Store (to a const threshold).

Updated generated remote execution protobuf code to latest from bazelbuild/remote-apis, and updated generation instructions.

### CODE REVIEW

File: bazel/cas/service.go [hunk 1]
```diff
@@ -98,37 +101,275 @@ func (s *casServer) FindMissingBlobs(
...
+	for _, digest := range req.GetBlobDigests() {
+		if err != nil {
+			break
 		}
+		sem <- struct{}{}
+		wg.Add(1)
+		go func(d *remoteexecution.Digest) {
+			// We hardcode support for empty data in snapshot/filer/checkouter.go, so never report it as missing
+			// Empty SHA can be used to represent working with a plain, empty directory, but can cause problems in Stores
+			if d.GetHash() == bazel.EmptySha {
```
Comment (ryancouto) [inline bazel/cas/service.go:131]: Should we check that d.SizeBytes is 0 too?

File: bazel/cas/service.go [hunk 2]
```diff
@@ -98,37 +101,275 @@ func (s *casServer) FindMissingBlobs(
...
+		if err != nil {
+			break
+		}
+		sem <- struct{}{}
+		wg.Add(1)
+		go func(r *remoteexecution.BatchUpdateBlobsRequest_Request) {
+			writeRes := &remoteexecution.BatchUpdateBlobsResponse_Response{Digest: r.GetDigest()}
+
+			// Return OK if we got a request to write the empty sha
+			if r.GetDigest().GetHash() == bazel.EmptySha {
```
Comment (ryancouto) [inline bazel/cas/service.go:209]: ditto for SizeByes

File: bazel/cas/service.go (hunk 1 above)
Comment (dgdonut) [reply bazel/cas/service.go:131]: We don't check for size in most existing places we check for EmptySha currently. I'm leaning towards skipping the size check everywhere we still do it (that would be like 2 places). It's kind of an undefined behavior on the part of the client, but on the server makes no practical difference as that data can never really exist internally.

File: bazel/cas/constants.go [hunk 3]
```diff
@@ -13,6 +13,10 @@ const (
 	// Default buffer sizes
 	DefaultReadCapacity = 1024 * 1024
 
+	// Batch parallelism for underlying store operations
+	// NOTE if service implementation is changed, retest with setting <= 5
+	BatchParallelism = 5
```
Comment (ryancouto) [inline bazel/cas/constants.go:19]: Why 5?

File: bazel/cas/service.go (hunk 1 above)
Comment (ryancouto) [reply bazel/cas/service.go:131]: Makes sense

**Review by ryancouto — APPROVED** [review]

File: bazel/cas/constants.go (hunk 3 above)
Comment (dgdonut) [reply bazel/cas/constants.go:19]: Good call, that's an arbitrary low value that will hopefully see some speedup but not cause issues. Adding a comment about it.

**Pull request merged**